gadget/install/partition.go

```diff
@@ -90,12 +90,12 @@ func createMissingPartitions(dl *gadget.OnDiskVolume, pv *gadget.LaidOutVolume)
 // returns a partitioning description suitable for sfdisk input and a
 // list of the partitions to be created.
 func buildPartitionList(dl *gadget.OnDiskVolume, pv *gadget.LaidOutVolume) (sfdiskInput *bytes.Buffer, toBeCreated []gadget.OnDiskStructure) {
-	sectorSize := dl.SectorSize
+	sectorSize := uint64(dl.SectorSize)
 
 	// Keep track what partitions we already have on disk
-	seen := map[quantity.Offset]bool{}
+	seen := map[uint64]bool{}
 	for _, s := range dl.Structure {
-		start := s.StartOffset / quantity.Offset(sectorSize)
+		start := uint64(s.StartOffset) / sectorSize
```

Comment:
quantity.Size|Offset are just types defined around uint64, so reverting back to uint64 does not affect the result

Reply:
I don't want to use quantity.Offset since the doc-comment for that type is: 
```go
// Offset describes the offset in bytes and is a thin wrapper around Size.
type Offset Size
```

The fact that the type is a unit in bytes means that here when we use it for sectors is "wrong"

Sure it all works the same because it's all just uint64s, but it is a bit misleading IMHO to use a type which says it is in bytes in places where the thing it is used for is not bytes

Reply:
If there's more use cases for sector offset, then perhaps we should rename Offset -> ByteOffset (leaving it as a wrapper around Size with IECString()) and introduce SectorOffset as a wrapper around uint4 with a n appropriate comment.
If that happens to be used in more than one place we could even have:
```
type SectorOffset struct {
    SectorSizeInBytes uint64
}
```
with some sector <-> byte conversion helpers. Still, only if there's more than one place where this could be used.

Reply:
Yeah I would be on board with something like that